reconcile/terraform_resources.py

```diff
@@ -469,10 +496,12 @@ def setup(dry_run, print_to_file, thread_pool_size, internal,
                           ocm_map=ocm_map)
     ts.dump(print_to_file, existing_dirs=working_dirs)
 
-    return ri, oc_map, tf, tf_namespaces
+    return ri, oc_map, tf, resource_specs
 
 
-def filter_tf_namespaces(namespaces, account_name):
+def filter_tf_namespaces(
+    namespaces: list[dict[str, Any]], account_name: Optional[str]
```

Comment:
Can we use `Iterable` and `Mapping` instead for the input types, or is it not possible here? There are a few other instances of this in the PR, but I'll avoid on commenting on them all.

Reply:
fixed

Reply:
I spoke to @Piojo about this earlier, but `Mapping` does in fact have `get`. When in doubt, we can always check [here](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes).

We have internal documentation on this topic [here](https://service.pages.redhat.com/dev-guidelines/docs/appsre/python-style-guide/#abstract-collection-types).

Reply:
i tried to apply it where possible but stopped at places where it would have leaked into ares of the code that are not within the scope of this PR.

expect more typehints in upcoming PRs but not this one

Reply:
Thanks for taking care of those updates!